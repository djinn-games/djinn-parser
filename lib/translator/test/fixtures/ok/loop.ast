{
  "type": "Program",
  "name": {
    "type": "Identifier",
    "name": "loop_sentence",
    "line": 1
  },
  "body": {
    "type": "Body",
    "sentences": [
      {
        "type": "LoopSentence",
        "sentences": [
          {
            "type": "IfSentence",
            "if": {
              "condition": {
                "type": "Literal",
                "line": 4,
                "dataType": "bool",
                "value": true
              },
              "consequent": [
                {
                  "type": "BreakSentence",
                  "line": 4
                }
              ],
              "alternates": [],
              "line": 4
            },
            "line": 4
          }
        ],
        "line": 3
      }
    ],
    "line": 2
  },
  "line": 1
}