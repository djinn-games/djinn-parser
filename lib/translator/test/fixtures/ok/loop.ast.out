{
    "type": "Program",
    "body": [
        {
            "type": "ForStatement",
            "init": null,
            "test": null,
            "update": null,
            "body": {
                "type": "BlockStatement",
                "body": [
                    {
                        "type": "IfStatement",
                        "test": {
                            "type": "Literal",
                            "value": true,
                            "raw": "true"
                        },
                        "consequent": {
                            "type": "BlockStatement",
                            "body": [
                                {
                                    "type": "BreakStatement",
                                    "label": null
                                }
                            ]
                        },
                        "alternate": {
                            "type": "BlockStatement",
                            "body": []
                        }
                    }
                ]
            }
        }
    ]
}
